### PR TITLE
ENH: Add max values comparison for floating point

### PR DESCRIPTION
**Pull request by howjmay**
SIMD intrinsics for comparing double/single precision has been
added. Now only NEON and SSE ones are added.

### CODE REVIEW

**Comment by charris** [comment]
See `doc/source/dev/development_workflow.rst` for the expected prefixes for the commit message.

**Review by seiko2plus — CHANGES_REQUESTED** [review]
Your code here is following x86 behavior rather than IEEE, since If "one" of the two arguments is NaN, the value of the other argument is returned.

Our SIMD interface tries not to differentiate between one architecture and another. in other words, it tries to provide the maximum possible performance for all architectures.

Therefore, I suggest the following:
- make `npyv_max_[f32, f64]` map directly to vmaxq_[f32, f64], _mm[,256,512]_max_[ps, pd], vec_max(VSX) which not oby the standards and also not provide a standardized behavior for NaN.   
- add new intrinics npyv_maxp_[f32, f64] that follow IEEE behavior.
  note: extra *p* for precise

You will also need to add a testing unit case for both intrinsic which requires the following:
- bring the new intrinics to Python world through SIMD module, check [_simd/_simd.dispatch.c.src](https://github.com/numpy/numpy/blob/main/numpy/core/src/_simd/_simd.dispatch.c.src)
- add the new test cases into [core/tests/test_simd.py](https://github.com/numpy/numpy/blob/main/numpy/core/tests/test_simd.py)

Thank you!

**Review by seiko2plus — CHANGES_REQUESTED** [review]
several fixes and improvements

**Comment by HowJMay** [comment]
Thank you @seiko2plus ! I learned a lot from your reviews. I have fixed them, please take a look.

File: numpy/core/src/common/simd/avx2/math.h [hunk 1]
```diff
@@ -37,4 +37,19 @@ NPY_FINLINE npyv_f32 npyv_square_f32(npyv_f32 a)
...
+NPY_FINLINE npyv_f32 npyv_maxp_f32(npyv_f32 a, npyv_f32 b) 
+{
+    npyv_f32 not_nan = _mm256_blendv_ps(_mm256_cmp_ps(a, a, _CMP_ORD_Q), a, b);
+    return _mm256_blendv_ps(_mm256_cmp_ps(a, b, _CMP_ORD_Q), _mm256_max_ps(a, b), not_nan);
+}
+NPY_FINLINE npyv_f64 npyv_maxp_f64(npyv_f64 a, npyv_f64 b) 
+{
+    npyv_f64 not_nan = _mm256_blendv_pd(_mm256_cmp_pd(a, a, _CMP_ORD_Q), a, b);
+    return _mm256_blendv_pd(_mm256_cmp_pd(a, b, _CMP_ORD_Q), _mm256_max_pd(a, b), not_nan);
+}
```
[the remark this answers is not in the document]
Comment (HowJMay) [reply numpy/core/src/common/simd/avx2/math.h]: May I ask where can I find this platform-specific information?

**Review by seiko2plus — APPROVED** [review]
LGTM, Thank you!

File: numpy/core/src/common/simd/vsx/math.h [hunk 2]
```diff
@@ -33,4 +33,10 @@ NPY_FINLINE npyv_f32 npyv_square_f32(npyv_f32 a)
 NPY_FINLINE npyv_f64 npyv_square_f64(npyv_f64 a)
 { return vec_mul(a, a); }
 
+// Maximum, natively mapping with no guarantees to handle NaN.
+#define npyv_max_f32 vec_max
+#define npyv_max_f64 vec_max
+#define npyv_maxp_f32 vec_max
```
Comment (seiko2plus) [inline numpy/core/src/common/simd/vsx/math.h:39]: `vec_max` supports IEEE

Comment (HowJMay) [reply numpy/core/src/common/simd/vsx/math.h:39]: Let me append the description for maxp just the same as other simd files.

**Pull request merged**